lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -454,3 +454,16 @@ <h4 class="message-title">${_('Your current verification will expire soon.')}</h
 <%static:require_module_async module_name="js/dateutil_factory" class_name="DateUtilFactory">
     DateUtilFactory.transform(iterationKey=".localized-datetime");
 </%static:require_module_async>
+
+
+<%static:require_module_async module_name="js/commerce/track_ecommerce_events" class_name="TrackECommerceEvents">
+  var link = $("#ecommerce_upgrade_link");
+  function trackEvent() {
+    TrackECommerceEvents.trackUpsellClick('course_dashboard_green', {
+      pageName: "course_dashboard",
+      linkType: "button",
+      linkCategory: "green_upgrade"
+    });
+  }
+  link.unbind('click.ecommerce_link').bind('click.ecommerce_link', trackEvent);
```

Comment:
Are you unbinding and then binding because this has been clicked already at this point? I think in jQuery you can also use `on()` and `off()`

Reply:
The issue is that the JS gets loaded twice on this page, which means that the event handler gets attached 2x, so every click would fire off two events. We weren't able to get on/off working - `.off()` by default removes all events (which we don't want to do, as there's other code attaching events to the click), and naming the event/passing it in didn't seem to work as the docs claimed.